src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,23 +2,139 @@
 
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
- * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
- * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
+ * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...) За
+ * бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private int size;
+    private int threshold;
+    private Node[] buckets;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (checkSize()) {
+            resize();
+        }
+        int index = getIndex(key);
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        if (buckets[index] == null) {
+            buckets[index] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+        Node<K, V> node = buckets[index];
+        if (checkBucketForExistKey(node, key)) {
```

Comment:
```suggestion
        if (checkBucketExistForKey(node, key)) {
```

Reply:
@kozhukhovsky you have at least two if checks inside this method, and two more if -else here. Why you can't insert data in the same method?